diff --git a/third_party/libopenjpeg20/j2k.c b/third_party/libopenjpeg20/j2k.c
index 14fc43089..1ae0baa42 100644
--- a/third_party/libopenjpeg20/j2k.c
+++ b/third_party/libopenjpeg20/j2k.c
@@ -5745,6 +5745,7 @@ static OPJ_BOOL opj_j2k_read_mct(opj_j2k_t *p_j2k,
     OPJ_UINT32 l_tmp;
     OPJ_UINT32 l_indix;
     opj_mct_data_t * l_mct_data;
+    OPJ_BOOL new_mct = OPJ_FALSE;
 
     /* preconditions */
     assert(p_header_data != 00);
@@ -5832,7 +5833,7 @@ static OPJ_BOOL opj_j2k_read_mct(opj_j2k_t *p_j2k,
         }
 
         l_mct_data = l_tcp->m_mct_records + l_tcp->m_nb_mct_records;
-        ++l_tcp->m_nb_mct_records;
+        new_mct = OPJ_TRUE;
     }
 
     if (l_mct_data->m_data) {
@@ -5864,6 +5865,9 @@ static OPJ_BOOL opj_j2k_read_mct(opj_j2k_t *p_j2k,
 
     l_mct_data->m_data_size = p_header_size;
 
+    if (new_mct) {
+        ++l_tcp->m_nb_mct_records;
+    }
     return OPJ_TRUE;
 }
 
